Source/ARTRealtime.m

```diff
@@ -707,7 +707,15 @@ - (ARTEventListener *)transitionSideEffects:(ARTConnectionStateChange *)stateCha
             channelStatus = stateChange.reason ? [ARTStatus state:ARTStateError info:stateChange.reason] : [self defaultError];
         }
         [self failQueuedMessages:channelStatus];
-        // For every Channel
+        
+        // If there's a channels.release() going on waiting on this channel
+        // to detach, failing it here would fire its event listener and
+        // immediately removing the channel from the channels dictionary, thus
+        // invalidating the iterator and causing a crashing.
+        //
+        // So fail later, when we're done using the iterator.
+        NSMutableArray<ARTRealtimeChannelInternal *> *toFail = [[NSMutableArray alloc] init];
```

Comment:
Please make pointers `const` wherever possible.

Reply:
Done at db0bc56, but if we're going to have this as a rule,

1. maybe we should write down a style guide or something,
2. why limit it to pointers? (Asking out if ignorance, I'm not familiar with the subtleties involved.)